### DISCUSSION TITLE
[Application Development] Using initiated withrawal transaction hash to proveMessage

### DISCUSSION THREAD

**Discussion by yilmof** [ethereum-optimism/developers#336] [Application Development]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Hello,

I am trying to prove and finalize a withrawal that has already been initiated. According to [the documentation](https://docs.optimism.io/builders/app-developers/tutorials/cross-dom-bridge-eth#withdraw-eth) I need to use a `ethers.providers.TransactionResponse` variable to pass to the `waitForMessageStatus` and `proveMessage` function.

```
const withdrawal = await messenger.withdrawETH(ethers.utils.parseEther('0.004269'))
await withdrawal.wait()

await messenger.waitForMessageStatus(withdrawal.hash, optimism.MessageStatus.READY_TO_PROVE)
```

When I try to pass just the hash string into waitForMessageStatus
`await messenger.waitForMessageStatus("tx hash string", optimism.MessageStatus.READY_TO_PROVE)`, 

I get
`error: withdrawal index 0 out of bounds. There are 0 withdrawals`

Is it possible to do this without having a `ethers.providers.TransactionResponse` variable?

Thanks

**Comment by 1195597989** [comment]
The transaction can be queried from the blockchain using the hash to reconstruct the required object, which you might find helpful.

**Reply by yilmof** [reply]
I tried doing this but I can't seem to get the correct `wait` property for `ethers.providers.TransactionResponse` (what is returned from the `withdrawETH()` function)

The definition of `ethers.providers.TransactionResponse`:
```
export interface TransactionResponse extends Transaction {
    hash: string;

    // Only if a transaction has been mined
    blockNumber?: number,
    blockHash?: string,
    timestamp?: number,

    confirmations: number,

    // Not optional (as it is in Transaction)
    from: string;

    // The raw transaction
    raw?: string,

    // This function waits until the transaction has been mined
    wait: (confirmations?: number) => Promise<TransactionReceipt>
};

```

I tried calling `withdrawETH` to see what the properties should be and this is what I get as the `wait` property function:
```
wait: function(confirms, timeout) {
        return __awaiter(_this, void 0, void 0, function() {
          var replacement, receipt;
          return __generator(this, function(_a) {
            switch (_a.label) {
              case 0:
                if (confirms == null)
                  confirms = 1;
                if (timeout == null)
                  timeout = 0;
                replacement = void 0;
                if (confirms !== 0 && startBlock != null)
                  replacement = {
                    data: tx.data,
                    from: tx.from,
                    nonce: tx.nonce,
                    to: tx.to,
                    value: tx.value,
                    startBlock
                  };
                return [4, this._waitForTransaction(tx.hash, confirms, timeout, replacement)];
              case 1:
                receipt = _a.sent();
                if (receipt == null && confirms === 0)
                  return [2, null];
                this._emitted["t:" + tx.hash] = receipt.blockNumber;
                if (receipt.status === 0)
                  logger.throwError("transaction failed", logger_1.Logger.errors.CALL_EXCEPTION, {
                    transactionHash: tx.hash,
                    transaction: tx,
                    receipt
                  });
                return [2, receipt];
            }
          });
        });
      }
```

**Reply by 1195597989** [reply]
The wait method in ethers.js is primarily designed to monitor transactions within the same Ethereum network. If you're handling a cross-chain transaction, the behavior of the wait method might not align with your expectations. This is because the transaction is processed on a different network before it ultimately reaches the Ethereum Mainnet.

**Reply by Chomtana** [reply]
Let's check what you have passed to l1SignerOrProvider while constructing the messenger object. Is the provider associated with that signer point to Ethereum Mainnet?

**Reply by yilmof** [reply]
Yes the l1SignerOrProvider in the messenger object is an ethers.Wallet with a private key and an ethers.providers.StaticJsonRpcProvider with an Ethereum mainnet rpc passed, as shown in the documentation

**Reply by yilmof** [reply]
The problem is that messenger.waitForMessageStatus and messenger.proveMessage won't run without an ethers.providers.TransactionResponse object passed that has all the required properties including the wait function

**Comment by Chomtana** [comment]
Is this on Optimism Sepolia?

**Reply by yilmof** [reply]
It is from OP Mainnet to Ethereum Mainnet
